src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -136,5 +137,30 @@ public static Set<Integer> parseShiftDays(String shiftDayString) {
                 .boxed().collect(Collectors.toSet());
     }
 
+    /**
+     * Parses {@code String specialisations} into a {@code Set<Specialisation>}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code specialisations} is invalid.
+     */
+    public static Set<seedu.address.model.person.Specialisation> parseSpecialisations(String specialisations)
+            throws ParseException {
+        requireNonNull(specialisations);
+        String trimmedSpecialisations = specialisations.trim();
+        if (trimmedSpecialisations.isEmpty()) {
+            return new HashSet<>();
+        }
+        String[] specialisationNames = trimmedSpecialisations.split(",");
+        Set<seedu.address.model.person.Specialisation> specialisationSet = new HashSet<>();
+        for (String specialisationName : specialisationNames) {
+            String specialisationNameTrimmed = specialisationName.trim();
+            if (!Specialisation.isValidSpecialisationName(specialisationNameTrimmed)) {
+                throw new ParseException(Specialisation.MESSAGE_CONSTRAINTS);
+            }
+            specialisationSet.add(new seedu.address.model.person.Specialisation(specialisationNameTrimmed));
```

Comment:
Might want to adjust the instantiation of `Specialisation`? `seedu.address.model.person.` is not needed here.